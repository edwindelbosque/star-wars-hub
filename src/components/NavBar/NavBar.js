import React from 'react'
import './NavBar.scss';
import { NavLink, Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavBar = ({ name, quote, skill, handleMovieChange, favorites, wipeUserInfo }) => {
  return (
    <nav>
      <header>
        <ul>
          <div className='user-info-nav'>
            <li className='user-info'>Name: <span>{name}</span></li>
            <li className='user-info'>Favorite Quote: <span>{quote}</span></li>
            <li className='user-info'>Skill Level: <span>{skill}</span></li>
          </div>
        </ul>
      </header>
      <footer>
        <ul>
          <div>
            <li>
              <NavLink to='/movies' className='nav-link' onClick={(e) => handleMovieChange()}>
                Movies
              </NavLink>
            </li>
            <li>
              <NavLink to='/favorites' className='nav-link'>
                Favorites: {favorites.length}
              </NavLink>
            </li>
          </div>
          <li>
            <Link to='/' className='nav-link-logout' onClick={() => wipeUserInfo()}>
              Log out
                        </Link>
          </li>
        </ul>
      </footer>
    </nav>
  )
}

NavBar.propTypes = {
  name: PropTypes.string,
  quote: PropTypes.string,
  skill: PropTypes.string
};

export default NavBar