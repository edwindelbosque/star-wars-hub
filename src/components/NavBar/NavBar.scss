nav {
    height: 110px;
    width: 100%;
    align-items: center;
    justify-content: center;
}

header {
    display: flex;
    height: 25px;
    color: black;
    background: rgb(38, 38, 38);
    align-items: center;
    border-bottom: 1px solid rgb(41, 41, 41);
    position: fixed;
    width: 100%;
    z-index: 5;
}

.user-info-nav {
    ul {
        width: 100%;

        li {
            margin-right: 300px;
        }
    }
}

.user-info {
    margin-right: 40px;
    font-size: 14px;
    font-weight: 400;
    color: rgb(111, 111, 111);

    span {
        font-weight: 600;
        color: #9b9b9b;
        font-size: 16px;
    }
}

ul {
    display: flex;
    color: rgb(199, 199, 199);
    padding: 0 130px;
    width: 100%;
    font-size: 18px;
    justify-content: space-between;
    align-items: center;
    text-align: center;
    list-style: none;

    div {
        display: flex;
    }
}

li {
    text-decoration: none;
}

footer {
    display: flex;
    height: 35px;
    background: #151515;
    font-weight: 600;
    position: fixed;
    width: 100%;
    top: 25px;
    z-index: 5;
}

.nav-link {
    text-decoration: none;
    color: rgb(198, 198, 198);
    font-size: 20px;
    transition: .2s;
    font-weight: 400;
    margin-right: 30px;
    border-bottom: solid 2px rgba(0, 0, 0, 0);
    padding-bottom: 4px;

    &:hover {
        color: rgb(255, 233, 70);
        transition: .2s;
        border-bottom: solid 2px rgb(255, 233, 70);
    }

    &:active {
        color: rgb(153, 137, 49);
        border-bottom: solid 2px rgb(143, 131, 38);
        transition: .2s;
    }
}

.nav-link-logout {
    text-decoration: none;
    color: rgba(255, 255, 255, 0.164);
    transition: .2s;
    font-size: 18px;
    font-weight: 400;

    &:hover {
        color: rgb(242, 80, 80);
        transition: .2s;
    }

    &:active {
        color: rgb(141, 46, 46);
        transition: .2s;
    }
}

.active {
    color: rgb(255, 233, 70);
    transition: .2s;
    border-bottom: solid 2px rgb(255, 233, 70);
}