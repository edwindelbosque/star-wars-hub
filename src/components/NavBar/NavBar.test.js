import React from 'react';
import { shallow, mount } from 'enzyme';
import NavBar from './NavBar';
import { NavLink } from 'react-router-dom'

describe('NavBar', () => {
  let wrapper;

  beforeEach(() => {
    const mockHandleMovieChange = jest.fn()
    wrapper = shallow(<NavBar 
      name={'Darth Vader'}
      quote={'I am your father.'}
      skill={'Expert'}
      favorites={0}
      handleMovieChange={mockHandleMovieChange}
    />)
  })
  it('component should match the snapshot', () => {
   
    expect(wrapper).toMatchSnapshot();
  });

  //attempted test but this is a Router issue
  // it('should call handleMovieChange when the movies link is clicked', () => {
  //   const mockHandleMovieChange = jest.fn()
  //   const wrapper = mount(<NavBar
  //     name={'Darth Vader'}
  //     quote={'I am your father.'}
  //     skill={'Expert'}
  //     favorites={0}
  //     handleMovieChange={mockHandleMovieChange} 
  //     />)

  //   wrapper.find(NavLink).first().simulate('click')

  //   expect(wrapper.props.handleMovieChange).toHaveBeenCalled();
  // })


})