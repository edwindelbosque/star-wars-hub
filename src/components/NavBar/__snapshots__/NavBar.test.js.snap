// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should match the snapshot 1`] = `
<nav>
  <header>
    <ul>
      <div
        className="user-info-nav"
      >
        <li
          className="user-info"
        >
          Name: 
          <span>
            Darth Vader
          </span>
        </li>
        <li
          className="user-info"
        >
          Favorite Quote: 
          <span>
            I am your father.
          </span>
        </li>
        <li
          className="user-info"
        >
          Skill Level: 
          <span>
            Expert
          </span>
        </li>
      </div>
    </ul>
  </header>
  <footer>
    <ul>
      <div>
        <li>
          <NavLink
            className="nav-link"
            onClick={[Function]}
            to="/movies"
          >
            Movies
          </NavLink>
        </li>
        <li>
          <NavLink
            className="nav-link"
            to="/favorites"
          >
            Favorites: 
          </NavLink>
        </li>
      </div>
      <li>
        <Link
          className="nav-link-logout"
          onClick={[Function]}
          to="/"
        >
          Log out
        </Link>
      </li>
    </ul>
  </footer>
</nav>
`;
